src/marqo/core/unstructured_vespa_index/unstructured_document.py

```diff
@@ -0,0 +1,180 @@
+
+        # vespa_embeddings may not return if it comes from a search result
+        embeddings_list = []
+        if self.vespa_embeddings:
+            try:
+                embeddings_list = list(self.vespa_embeddings["blocks"].values())
+            except (KeyError, AttributeError, TypeError):
+                raise VespaDocumentParsingError(f"No embeddings found in the document _id = {self.marqo__id}")
+
+        if self.vespa_chunks and embeddings_list:
+            if not len(self.vespa_chunks) == len(embeddings_list):
+                raise ValueError(f"Number of chunks and embeddings do not match for document _id= {self.marqo__id}")
+            marqo_document[index_constants.MARQO_DOC_TENSORS] = dict()
+            for chunk, embedding in zip(self.vespa_chunks, embeddings_list):
+                if "::" not in chunk:
+                    raise ValueError(f"Chunk {chunk} does not have a field_name::content format")
+                field_name, content = chunk.split("::", 2)
+                if field_name not in marqo_document[index_constants.MARQO_DOC_TENSORS]:
+                    marqo_document[index_constants.MARQO_DOC_TENSORS][field_name] = dict()
+                    marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][index_constants.MARQO_DOC_CHUNKS] = []
+                    marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][
+                        index_constants.MARQO_DOC_EMBEDDINGS] = []
+                marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][index_constants.MARQO_DOC_CHUNKS].append(
+                    content)
+                marqo_document[index_constants.MARQO_DOC_TENSORS][field_name][
+                    index_constants.MARQO_DOC_EMBEDDINGS].append(embedding)
+
+        if self.vespa_multimodal_params:
+            marqo_document[unstructured_common.MARQO_DOC_MULTIMODAL_PARAMS] = dict()
+            for multimodal_field_name, serialized_multimodal_params in self.vespa_multimodal_params.items():
+                marqo_document[unstructured_common.MARQO_DOC_MULTIMODAL_PARAMS][multimodal_field_name] = \
+                    json.loads(serialized_multimodal_params)
+
+        if return_highlights and self.match_features:
+            marqo_document[index_constants.MARQO_DOC_HIGHLIGHTS] = self._extract_highlights()
+
+        return marqo_document
+
+    def _extract_highlights(self):
+        if not self.match_features:
+            raise ValueError("No match features found in the document")
+        try:
+            chunk_index: int = int(list(self.match_features[f"closest({unstructured_common.VESPA_DOC_EMBEDDINGS})"] \
+                                            ["cells"].keys())[0])
+        except KeyError:
+            raise ValueError("No match features found in the document")
+        field_name, content = self.vespa_chunks[chunk_index].split("::", 2)
+        return {field_name: content}
+
+
+class UnstructuredIndexDocument(BaseModel):
```

Comment:
I find class names a bti confusing here. How about `UnstructuredVespaDocument` and `UnstructuredVespaDocumentFields` (unchanged)?